__init__.py

```diff
@@ -43,8 +49,8 @@ def _setup(self, force=False):
                 # String might be some rubbish (like '')
                 portnumber = 0
```

Comment:
I know this is original code, but what do you think, does it make much sense to set the port to 0? This would cause the connection to fail to 99% of HA installs.

You make a great check at line 37-42 which makes sense. If we are changing the base anyways, would you mind thinking out a similar check for the port?

Reply:
Okay, how would you like to handle this? Set it to standart port or add it as error message?

Reply:
I think because you error out on the IP and token, it should be consistent and error out

Reply:
Same message or divide?

Reply:
That's a good point, it should be a different message so the user knows what's wrong. If I were making this PR, I would have an error message for each IP, token and port so that the user has a clear indication of where the problem lies

Reply:
Ok, give me few minutes.

Reply:
Done, I used same message, only added value to it. All languages are updated with english. Is it OK this way?

Should I rebase?

Reply:
No need to rebase. I fixed the German translation. 

As we mentioned in the other PR, we prefer to have nothing in other languages as opposed to a file that contains a language other than specified. I.E.: if its `it-it` it should only have Italian in the file

Reply:
And how would you knot that there is need for update translation?

Reply:
If the dialog/translation file is missing, there is a different place where the Mycroft team handles translations. There are contributors that go to a different location to determine when something needs translating. I am not involved in that so I don't know much more than that. Their process is based on the files not being present

Reply:
Yes but when I just update already present file, it will not get recognized.

Reply:
Hey, for any string that hasn't already been translated, translate.mycroft.ai will show it as not yet complete. This includes files that were already present if the strings inside them changed. Eg if I have a `test.voc` file containing:
```
test
testing
```
and these have been translated already, then I change it to:
```
test
testing
tested
```
Then translate.mycroft.ai should indicate that only "tested" needs to be translated.

Let me know if I misunderstood the question.

Reply:
@krisgesling and does it also work in opposite direction? When I update string in non english, will it be recognized or just rewroted in next update from translation platform?

Reply:
I'd have to check, but fairly sure it's limited to English initiating these processes at the moment. This does also mean that future updates from the translation platform will overwrite changes that are only made directly in Skills.

Reply:
@krisgesling could this behavior be changed to scan also all languages or make it possible to manually load translations to translate platform?

Reply:
Unfortunately two-way sync is always a more difficult problem. The big challenge is knowing which is the most up to date. 

For example, if a PR updates the langauge files in a Skill and in the same period a user on translate.mycroft.ai updates the translations there. When you sync do you take the Skill or the translation version as being correct?

If you want to look into it you can see the scripts that do the importing and exporting here: https://github.com/MycroftAI/mycroft-translate